src/Features/CSharp/Portable/Snippets/ConsoleSnippetProvider.cs

```diff
@@ -0,0 +1,179 @@
+                             node.IsKind(SyntaxKind.ArgumentList) ||
+                             node.IsKind(SyntaxKind.RecordDeclaration) ||
+                             node.IsKind(SyntaxKind.ObjectCreationExpression) ||
+                             node.IsKind(SyntaxKind.SwitchExpression));
+
+            if (isDirectlyInUndesirableLocation)
+            {
+                return false;
+            }
+
+            var isExpressionInVariable = token.GetAncestors<SyntaxNode>()
+                .Any(node => node.IsKind(SyntaxKind.ParenthesizedLambdaExpression) || node.IsKind(SyntaxKind.AnonymousMethodExpression));
+            var isInVariableDeclaration = token.GetAncestors<SyntaxNode>().Any(node => node.IsKind(SyntaxKind.VariableDeclaration));
+
+            if (isInVariableDeclaration && !isExpressionInVariable)
+            {
+                return false;
+            }
+
+            var isInsideMethod = token.GetAncestors<SyntaxNode>()
+               .Any(node => node.IsKind(SyntaxKind.MethodDeclaration) ||
+                            node.IsKind(SyntaxKind.ConstructorDeclaration) ||
+                            node.IsKind(SyntaxKind.LocalFunctionStatement) ||
+                            node.IsKind(SyntaxKind.AnonymousMethodExpression) ||
+                            node.IsKind(SyntaxKind.ParenthesizedLambdaExpression));
+
+            var isInNamespace = token.GetAncestors<SyntaxNode>()
+                .Any(node => node.IsKind(SyntaxKind.NamespaceDeclaration) ||
+                             node.IsKind(SyntaxKind.FileScopedNamespaceDeclaration));
+
+            if (isInNamespace && !isInsideMethod)
+            {
+                return false;
+            }
+
+            if (!isInsideMethod && !context.IsGlobalStatementContext)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override string GetSnippetDisplayName()
+        {
+            return FeaturesResources.Write_to_the_Console;
+        }
+
+        protected override async Task<TextChange> GenerateSnippetTextChangeAsync(Document document, TextSpan span, int tokenSpanStart, int tokenSpanEnd, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = tree.FindTokenOnLeftOfPosition(tokenSpanEnd, cancellationToken);
+            var declaration = GetAsyncSupportingDeclaration(token);
+            var isAsync = generator.GetModifiers(declaration).IsAsync;
+            SyntaxNode? invocation;
+            invocation = isAsync
+                ? generator.ExpressionStatement(generator.AwaitExpression(generator.InvocationExpression(
+                    generator.MemberAccessExpression(generator.MemberAccessExpression(generator.IdentifierName("Console"),
```

Comment:
What if there is no `using System;` at the top of the file? Does this need to be fully qualified with a simplifier annotation? Or does this method need to be able to return multiple TextChanges (or a changed document) and add the using itself? (Can completion do that?)